snuba/utils/streams/kafka.py

```diff
@@ -58,12 +58,30 @@ def __init__(self, state: KafkaConsumerState):
         self.__state = state
 
 
-@dataclass(frozen=True)
```

Comment:
dataclasses don't support `__slots__` with default arguments. 😞 

I think it's more important that this retains a lightweight memory footprint and has a default value than using dataclasses without the default argument.